File: check_users.py
#!/usr/bin/env python3
"""
Check what users exist in the system
"""
import os
from supabase import create_client
from dotenv import load_dotenv
from pathlib import Path

# Try loading from .env files
home_dir = Path.home()
env_file = home_dir / '.env'
if env_file.exists():
    load_dotenv(env_file)

# Or from frontend/.env.local
frontend_env = Path(__file__).parent / 'frontend' / '.env.local'
if frontend_env.exists():
    load_dotenv(frontend_env)

SUPABASE_URL = os.getenv('NEXT_PUBLIC_SUPABASE_URL')
SUPABASE_KEY = os.getenv('NEXT_PUBLIC_SUPABASE_ANON_KEY')

print(f"URL: {SUPABASE_URL}")
print(f"KEY: {SUPABASE_KEY[:20] if SUPABASE_KEY else 'NOT FOUND'}...")

if not SUPABASE_URL or not SUPABASE_KEY:
    print("\n❌ Cannot find Supabase credentials")
    print("Looking for env variables...")
    print(f"  NEXT_PUBLIC_SUPABASE_URL: {os.getenv('NEXT_PUBLIC_SUPABASE_URL')}")
    print(f"  NEXT_PUBLIC_SUPABASE_ANON_KEY: {os.getenv('NEXT_PUBLIC_SUPABASE_ANON_KEY')}")
    exit(1)

try:
    supabase = create_client(SUPABASE_URL, SUPABASE_KEY)
    
    # Get all users
    response = supabase.table('users').select('id,email,role,full_name').execute()
    users = response.data if response.data else []
    
    print(f"\n✅ Found {len(users)} users:")
    for user in users:
        print(f"   ID: {user['id'][:8]}...")
        print(f"      Email: {user['email']}")
        print(f"      Name: {user['full_name']}")
        print(f"      Role: {user['role']}")
        print()
    
except Exception as e:
    print(f"❌ Error: {e}")
    exit(1)

File: check-users.ts
import { createClient } from "@supabase/supabase-js";

const SUPABASE_URL = process.env.SUPABASE_URL || process.env.NEXT_PUBLIC_SUPABASE_URL || "";
const SUPABASE_SERVICE_ROLE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY || "";

if (!SUPABASE_URL || !SUPABASE_SERVICE_ROLE_KEY) {
  console.error("{0} Missing SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY environment variables", String.fromCharCode(0x274C));
  console.error("   Set them in your .env file or export them before running this script.");
  process.exit(1);
}

const supabaseAdmin = createClient(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY);

async function checkUsers() {
  console.log("{0} Checking users in Supabase...\n", String.fromCharCode(0x1F50D));

  const { data: users, error } = await supabaseAdmin
    .from("users")
    .select("id, email, username, full_name, role, is_active")
    .order("created_at", { ascending: false });

  if (error) {
    console.error("{0} Error fetching users:", String.fromCharCode(0x274C), error.message);
    return;
  }

  if (!users || users.length === 0) {
    console.log("{0}  No users found in the database!", String.fromCharCode(0x26A0));
    return;
  }

  console.log("{0} Found " + users.length + " users:\n", String.fromCharCode(0x2705));
  console.table(users);
}

checkUsers();